runtime/parser/expression.go

```diff
@@ -590,7 +632,8 @@ func defineLessThanOrTypeArgumentsExpression() {
 
 			var argumentsStartPos ast.Position
 
-			(func() {
+			// Ignore the error
+			_ = func() error {
```

Comment:
We can now use the error here to determine if parsing as an invocation succeeded and can remove `isInvocation` now (basically `isInvocation` iff `err != nil`)